File: data/code_pairs/fortran/593.f90
program implicit_input_output
    integer :: i, j
    real :: a(2, 3)

    ! Fill the matrix
    do i = 1, 2
        do j = 1, 3
            a(i, j) = 10*i + j
        end do
    end do

    ! Print the entire matrix
    do j = 1, 3
        do i = 1, 2
            print *, a(i, j),
        end do
    end do
    print *, "" ! New line

    ! Print the first row
    do j = 1, 3
        print *, a(1, j),
    end do
    print *, "" ! New line

    ! Print the first column
    do i = 1, 2
        print *, a(i, 1),
    end do
    print *, "" ! New line

end program implicit_input_output

File: data/code_pairs/cpp/593.cpp
#include <iostream>
#include <vector>

int main() {
    const int rows = 2;
    const int cols = 3;
    std::vector<std::vector<float>> a(rows, std::vector<float>(cols));

    // Fill the matrix
    for (int i = 0; i < rows; ++i) {
        for (int j = 0; j < cols; ++j) {
            a[i][j] = 10 * (i + 1) + (j + 1);
        }
    }

    // Print the entire matrix in the specified order
    for (int j = 0; j < cols; ++j) {
        for (int i = 0; i < rows; ++i) {
            std::cout << a[i][j] << " ";
        }
    }
    std::cout << std::endl;

    // Print the first row
    for (int j = 0; j < cols; ++j) {
        std::cout << a[0][j] << " ";
    }
    std::cout << std::endl;

    // Print the first column
    for (int i = 0; i < rows; ++i) {
        std::cout << a[i][0] << " ";
    }
    std::cout << std::endl;

    return 0;
}